components/calendar/calendar.js

```diff
@@ -0,0 +1,474 @@
+					return html`
+						<td>
+							<d2l-calendar-date
+								date=${day.date}
+								month=${day.month}
+								year=${day.year}
+								?focused=${day.focused}
+								?other-month=${day.otherMonth}
+								?selected=${day.selected}>
+							</d2l-calendar-date>
+						</td>`;
+				});
+
+				return html`
+					<tr>${weekHtml}</tr>
+				`;
+			});
+		}
+		const heading = `${this._monthNames[this._shownMonth]} ${this._shownYear}`;
+		return html`
+			<div aria-labelledby="${this._dialogLabelId}" class="d2l-calendar">
+				<table summary="${this.summary}" role="grid">
+					<thead>
+						<tr class="d2l-calendar-title">
+							<td>
+								<d2l-button-icon @click="${this._showPrevMonth}" text="${this._computeText(this._prevMonth)}" icon="tier1:chevron-left"></d2l-button-icon>
+							</td>
+							<th colspan="5">
+								<h2 class="d2l-heading-4" aria-live="polite" id="${this._dialogLabelId}">${heading}</h2>
+							</th>
+							<td>
+								<d2l-button-icon @click="${this._showNextMonth}" text="${this._computeText(this._nextMonth)}" icon="tier1:chevron-right"></d2l-button-icon>
+							</td>
+						</tr>
+					</thead>
+					<tbody>
+						<tr>
+							${weekdayHeaders}
+						</tr>
+						${dayRows}
+					</tbody>
+				</table>
+				<slot></slot>
+			</div>
+		`;
+	}
+
+	_changeFocusDate(numDays) {
+		this._getNodeAndRemoveFocus();
+
+		const d =  new Date(this._focusDate.year, this._focusDate.month, this._focusDate.date);
+		d.setDate(d.getDate() + numDays);
+
+		this._focusDate = {
+			date: d.getDate(),
+			month: d.getMonth(),
+			year: d.getFullYear()
+		};
+
+		this._keyboardTriggeredMonthChange = true;
+		if (this._focusDate.year < this._shownYear || this._focusDate.month < this._shownMonth) {
+			this._showPrevMonth();
+		} else if (this._focusDate.month > this._shownMonth || (this._shownMonth === 11 && this._focusDate.month === 0)) {
+			this._showNextMonth();
+		} else {
+			this._getNodeAndAddFocus(this._keyboardTriggeredMonthChange);
+			this._keyboardTriggeredMonthChange = false;
+		}
+	}
+
+	_checkIfDatesEqual(month, date, year, comparisonDate) {
+		return comparisonDate.month === month
+			&& comparisonDate.date === date
+			&& comparisonDate.year === year;
+	}
+
+	_computeText(month) {
+		return this.localize('show', {month: this._monthNames[month]});
+	}
+
+	_createDateObject(date, prevMonth, nextMonth) {
+		let month = this._shownMonth;
+		let year = this._shownYear;
+		if (prevMonth) {
+			month = this._prevMonth;
+			if (this._shownMonth === 0) year = this._shownYear - 1;
+		} else if (nextMonth) {
+			month = this._nextMonth;
+			if (this._shownMonth === 11) year = this._shownYear + 1;
+		}
+
+		return {
+			date: date,
+			month: month,
+			year: year,
+			otherMonth: prevMonth || nextMonth,
+			selected: this._checkIfDatesEqual(month, date, year, this._selectedDate),
+			focused: this._checkIfDatesEqual(month, date, year, this._focusDate)
+		};
+	}
+
+	_generateDaysInMonth() {
```

Comment:
This and `_getNumberOfDaysInMonth` seem like they could just be utility functions outside of the class. The more we can move out of the class itself, the smaller the classes are that get instantiated over and over by the browser.